guice/src/main/java/com/flipkart/gjex/grpc/interceptor/FilterInterceptor.java

```diff
@@ -82,58 +84,36 @@ public void registerFilters(List<Filter> filters, List<BindableService> services
             filtersMap.get(call.getMethodDescriptor().getFullMethodName().toLowerCase());
         Metadata forwardHeaders = new Metadata();
-        if (filterReferences == null || filterReferences.isEmpty()){
-            return new SimpleForwardingServerCallListener<ReqT>(next.startCall(
-                new SimpleForwardingServerCall<ReqT, RespT>(call) {
+        if (grpcFilterReferences == null || grpcFilterReferences.isEmpty()){
+            return new SimpleForwardingServerCallListener<Req>(next.startCall(
+                new SimpleForwardingServerCall<Req, Res>(call) {
                 }, headers)) {
             };
         }
-        List<Filter> filters = filterReferences.stream().map(Filter::getInstance).collect(Collectors.toList());
-        for (Filter filter : filters) {
-            try {
-                if (filter != null) {
-                    ServerRequestParams serverRequestParams =
-                        ServerRequestParams.builder()
-                            .clientIp(call.getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR).toString())
-                            .methodName(call.getMethodDescriptor().getFullMethodName().toLowerCase())
-                            .build();
-                    filter.doFilterRequest(serverRequestParams, headers);
-                    for (Metadata.Key key : filter.getForwardHeaderKeys()) {
-                        Object value = headers.get(key);
-                        if (value != null) {
-                            forwardHeaders.put(key, value);
-                        }
-                    }
-                }
-            } catch (StatusRuntimeException se) {
-                call.close(se.getStatus(), se.getTrailers()); // Closing the call and not letting it to proceed further
-                return new ServerCall.Listener<ReqT>() {
-                };
-            }
-        }
-
+        List<GjexGrpcFilter> grpcFilters = grpcFilterReferences.stream().map(GjexGrpcFilter::getInstance).collect(Collectors.toList());
         Context contextWithHeaders = forwardHeaders.keys().isEmpty() ? null : Context.current().withValue(GJEXContext.getHeadersKey(), forwardHeaders);
 
-        ServerCall.Listener<ReqT> listener = null;
-        listener = next.startCall(new SimpleForwardingServerCall<ReqT, RespT>(call) {
+        ServerCall.Listener<Req> listener = null;
+        listener = next.startCall(new SimpleForwardingServerCall<Req, Res>(call) {
             @Override
-            public void sendMessage(final RespT response) {
+            public void sendMessage(final Res response) {
                 Context previous = attachContext(contextWithHeaders);   // attaching headers to gRPC context
                 try {
-                    filters.forEach(filter -> filter.doProcessResponse((GeneratedMessageV3) response));
+                    grpcFilters.forEach(filter -> filter.doProcessResponse(ResponseParams.builder().response(response).build()));
```

Comment:
why wrap? what is the usecase?

Reply:
ResponseParams is a common class used for both http and grpc responses. so wrapped the grpc "response"